src/hotspot/share/ci/ciMethodData.cpp

```diff
@@ -123,12 +130,15 @@ void ciMethodData::prepare_metadata() {
 
 void ciMethodData::load_remaining_extra_data() {
   MethodData* mdo = get_MethodData();
-  MutexLocker ml(mdo->extra_data_lock());
+
+  // Lock to read ProfileData, and ensure lock is not unintentionally broken by a safepoint
+  MutexLocker ml(mdo->extra_data_lock(), Mutex::_no_safepoint_check_flag);
```

Comment:
Is there anyway to have MutexLocker take care of verifying that the there's no safepoint? It would be nice to replace:

```
MutexLocker ml();
NoSafepointVerifier no_safepoint;
```

by: 

```
MutexLocker ml();
```

only.

Reply:
@fisk @tkrodriguez what do you suggest for that?

Reply:
Should I create a wrapper object for it? Maybe a `MutexLockerAndNoSafepointVerifier`? Would I place that simply in `methodData.hpp`? Or make it available more widely?

Reply:
But in this specific instance I keep them separate, because I need to pass the `NoSafepointVerifier` down, so that I can pass it to the `PauseNoSafepointVerifier`.